### PR TITLE
feat: bump peer dep styled-components to v5.1.0 minimum

### PR DESCRIPTION
**Pull request by geotrev**
## Description

Updates `styled-components` peer dependency version.

- Component packages bumped to `^5.1.0`.
- `react-theming` bumped to `^4.2.0 || ^5.1.0` (from `^4.2.0 || ^5.0.0`)
- Updates `migration.md` with breaking change notes.

## Detail

No component code changes necessary at this time.

## Checklist

- [ ] ~~:ok_hand: design updates will be Garden Designer approved (add the designer as a reviewer)~~
- [ ] ~~:globe_with_meridians: demo is up-to-date (`npm start`)~~
- [ ] ~~:arrow_left: renders as expected with reversed (RTL) direction~~
- [ ] ~~:metal: renders as expected with Bedrock CSS (`?bedrock`)~~
- [ ] ~~:guardsman: includes new unit tests. Maintain existing coverage (always >= 96%)~~
- [ ] ~~:wheelchair: tested for WCAG 2.1 AA accessibility compliance~~
- [ ] ~~:memo: tested in Chrome, Firefox, Safari, and Edge~~

### CODE REVIEW

File: docs/migration.md [hunk 1]
```diff
@@ -4,6 +4,13 @@
 
 ### Breaking Changes
 
+#### All Packages
+
+- Garden's React packages now require a minimum peer dependency of
+  `styled-components` at `v6.0.0`.
+  - Any custom styled components should follow the styled components v6
+    [migration requirements](https://styled-components.com/docs/faqs#what-do-i-need-to-do-to-migrate-to-v6).
```
Comment (jzempel) [inline docs/migration.md:12]: This is not correct. When v9 ships, the minimum for `styled-components` will be `v5.1.0`. After v9, we'll chip away at renaming transient styled props, and allow a `^5.1.0 || ^6.00` dependency range on a package-by-package basis. But there is no need to spell that out here. See sequence doc for details.

**Review by jzempel — CHANGES_REQUESTED** [review]
The `theming` package is missing here. It should be updated to `^4.2.0 || ^5.1.0`

File: docs/migration.md (hunk 1 above)
Comment (geotrev) [reply docs/migration.md:12]: I misinterpreted the sequence. Sounds good.

Comment (geotrev) [reply docs/migration.md:12]: Both the plan sequence and ticket say the v5 range is `5.0.0` for the v5 range in `react-theming` - probably just a typo. I'll update the branch & ticket to say `5.1.0`.

Comment (jzempel) [reply docs/migration.md]: Yeah, we need `5.1.0` in order to make the jump to tran$ient props before they are forced in `6.0.0`.

**Review by jzempel — APPROVED** [review]

**Pull request merged**